rage/vector.hpp

```diff
@@ -27,22 +27,133 @@ namespace rage
 		T data[3];
 		struct { T x, y, z; };
 
-		vector3(T x, T y, T z) :
-			x(x),
+		vector3(T x, T y, T z) noexcept : x(x),
```

Comment:
You've declared the vector as templateable which is fine as we did that before but have now added the noexcept keyword. If any of the constructors of T throw an exception, this would violate the noexcept guarantee. This would be fine if we only use it for floats, but if that's the case then why don't we just strongly type?